plugins/woocommerce/includes/abstracts/abstract-wc-data.php

```diff
@@ -161,9 +161,6 @@ public function __clone() {
 		if ( ! empty( $this->meta_data ) ) {
 			foreach ( $this->meta_data as $array_key => $meta ) {
 				$this->meta_data[ $array_key ] = clone $meta;
-				if ( ! empty( $meta->id ) ) {
-					$this->meta_data[ $array_key ]->id = null;
-				}
```

Comment:
I don't think it's safe to remove this because of the impact it has on cloned instances:

```php
$copy = clone $product;
$copy->save_meta_data();
```

This change makes it so that it will update the existing meta entries instead of creating new ones. I would _absolutely_ agree that the existing behavior is wrong, however, it has been this way for eight years.